File: typescript/easy/merge-strings-alternately.ts
function mergeAlternately(word1: string, word2: string): string {
    let result: string[] = [];
    const minLen = Math.min(word1.length, word2.length);

    for (let i = 0; i < minLen; i++) {
        result.push(word1[i]);
        result.push(word2[i]);
    }

    if (word1.length > minLen) {
        result.push(word1.slice(minLen));
    } else if (word2.length > minLen) {
        result.push(word2.slice(minLen));
    }

    return result.join("");
}

File: csharp/easy/merge-strings-alternately.cs
string MergeAlternately(string word1, string word2) {
        StringBuilder result = new StringBuilder();
        int minLen = Math.Min(word1.Length, word2.Length);

        for (int i = 0; i < minLen; i++) {
            result.Append(word1[i]);
            result.Append(word2[i]);
        }

        if (word1.Length > minLen) {
            result.Append(word1.Substring(minLen));
        } else if (word2.Length > minLen) {
            result.Append(word2.Substring(minLen));
        }

        return result.ToString();
    }
